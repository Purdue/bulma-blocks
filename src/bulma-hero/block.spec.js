jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    clickRadio,
    blockStartup,
    selectOption
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Hero', blockName: 'bulma-blocks/hero'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe( 'ðŸ”¬ Side Panel Settings', () => {
        describe( 'ðŸ”¬ Radio Settings', () => {
            test( 'ðŸ”Ž Padding Size Options', async () => {
                await blockStartup(block)

                // select medium padding
                await clickRadio('Padding Size', 'is-medium')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"option":"is-medium"')).toBe(true)

                // select large padding
                await clickRadio('Padding Size', 'is-large')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"option":"is-large"')).toBe(true)

                // select small padding
                await clickRadio('Padding Size', 'sm')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"option":"is-medium"')).toBe(false)
                expect( editedContent.includes('"option":"is-large"')).toBe(false)
            })
        })
        describe( 'ðŸ”¬ Dropdown Settings', () => {
            test( 'ðŸ”Ž Background Color Options', async () => {
                await blockStartup(block)

                // black background option select
                await selectOption('Background Color', 'has-background-black')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                
                // light grey background option select
                await selectOption('Background Color', 'has-background-grey-lighter')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-grey-lighter"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()
                

                // dark grey background option select
                await selectOption('Background Color', 'has-background-black-ter')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black-ter"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()


                // no background option select (DEFAULT)
                await selectOption('Background Color', '')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black"')).toBe(false)
                expect( editedContent.includes('"bgColor":"has-background-grey-lighter"')).toBe(false)
                expect( editedContent.includes('"bgColor":"has-background-black-ter"')).toBe(false)
                expect( editedContent ).toMatchSnapshot()
            })

        })
    })

})