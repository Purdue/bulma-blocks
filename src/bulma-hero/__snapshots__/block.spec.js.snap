// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Bulma - Hero Block 🔎 Block should be available. 1`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\"} -->
<div class=\\"hero sm \\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 1`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\",\\"bgColor\\":\\"has-background-black\\"} -->
<div class=\\"hero sm has-background-black\\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 2`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\",\\"bgColor\\":\\"has-background-grey-lighter\\"} -->
<div class=\\"hero sm has-background-grey-lighter\\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 3`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\",\\"bgColor\\":\\"has-background-black-ter\\"} -->
<div class=\\"hero sm has-background-black-ter\\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 4`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\"} -->
<div class=\\"hero sm \\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Padding Size Options 1`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"is-medium\\"} -->
<div class=\\"hero is-medium \\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Padding Size Options 2`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"is-large\\"} -->
<div class=\\"hero is-large \\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;

exports[`🔬 Bulma - Hero Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Padding Size Options 3`] = `
"<!-- wp:bulma-blocks/hero {\\"option\\":\\"sm\\"} -->
<div class=\\"hero sm \\"><div class=\\"hero-body\\"></div></div>
<!-- /wp:bulma-blocks/hero -->"
`;
